src/components/InfiniteGrid.js

```diff
@@ -0,0 +1,224 @@
+var React = require('react/addons');
+var PureRenderMixin = React.addons.PureRenderMixin;
+
+/**
+ * Inspired from: https://github.com/ggordan/react-infinite-grid
```

Comment:
A lot of scroll/lazycallback inspiration came from it, but I did remove a ton of logic, updated broken items and overall association to work for Kitematic. I don't mind adding the MIT license if needed